gluoncv/data/mscoco/detection.py

```diff
@@ -195,3 +203,45 @@ def _check_load_bbox(self, coco, entry):
                 # dummy invalid labels if no valid objects are found
                 valid_objs.append([-1, -1, -1, -1, -1])
         return valid_objs
+
+class COCODetectionDALI(object):
+    """DALI partial pipeline with COCO Reader and loader. To be passed as
+    a parameter of a DALI transform pipeline.
+
+    Parameters
+    ----------
+    num_shards: int
+         DALI pipeline arg - Number of pipelines used, indicating to the reader
+         how to split/shard the dataset.
+    shard_id: int
+         DALI pipeline arg - Shard id of the pipeline must be in [0, num_shards).
+    file_root
+        Directory containing the COCO dataset.
+    annotations_file
+        The COCO annotation file to read from.
+    """
+    def __init__(self, num_shards, shard_id, file_root, annotations_file):
+        self.input = ops.COCOReader(
+            file_root=file_root,
+            annotations_file=annotations_file,
+            skip_empty=True,
+            shard_id=shard_id,
+            num_shards=num_shards,
+            ratio=True,
+            ltrb=True,
+            shuffle_after_epoch=True)
+
+        self.decode = ops.HostDecoder(device="cpu", output_type=types.RGB)
+
+    def __call__(self):
+        """Returns three DALI graph nodes: inputs, bboxes, labels.
+        To be called in `define_graph`.
+        """
+        inputs, bboxes, labels = self.input(name="Reader")
+        images = self.decode(inputs)
+        return (images, bboxes, labels)
+    def size(self):
+        """Returns size of COCO dataset
+        """
+        coco_size = 118287
```

Comment:
anyway to parse the annotation_file? we have users who use their custom dataset in coco compatible format

Reply:
is COCOReader able to return the number of images?

Reply:
I see, yes COCOReader is able to return it, but we have to build the whole DALI pipeline then destroy i
It's not really clean.. but would be actually faster than the default Python parser that takes 15s to parse vs 0.5s for DALI.

Reply:
One feature on DALI roadmap is to have standalone ops that can be directly run in Python, in the meantime I'll add this "1-op micro Pipeline"

Reply:
This and other review comments addressed in https://github.com/dmlc/gluon-cv/pull/807/commits/3b6258c4d2fa06cb304d14acf395abfe1620b46e